comp/logs/agent/agentimpl/agent_core_init.go

```diff
@@ -58,13 +58,70 @@ func (a *logAgent) SetupPipeline(processingRules []*config.ProcessingRule, wmeta
 	// setup the launchers
 	lnchrs := launchers.NewLaunchers(a.sources, pipelineProvider, a.auditor, a.tracker)
 
+	a.addLauncherInstances(lnchrs, wmeta, integrationsLogs, fingerprintConfig)
+
+	a.schedulers = schedulers.NewSchedulers(a.sources, a.services)
+	a.destinationsCtx = destinationsCtx
+	a.pipelineProvider = pipelineProvider
+	a.launchers = lnchrs
+	a.diagnosticMessageReceiver = diagnosticMessageReceiver
+}
+
+// buildEndpoints builds endpoints for the logs agent, either HTTP or TCP,
+// dependent on configuration and connectivity
+func buildEndpoints(coreConfig model.Reader) (*config.Endpoints, error) {
+	httpConnectivity := config.HTTPConnectivityFailure
+	if endpoints, err := config.BuildHTTPEndpointsWithVectorOverride(coreConfig, intakeTrackType, config.AgentJSONIntakeProtocol, config.DefaultIntakeOrigin); err == nil {
+		httpConnectivity = http.CheckConnectivity(endpoints.Main, coreConfig)
+	}
+	return config.BuildEndpointsWithVectorOverride(coreConfig, httpConnectivity, intakeTrackType, config.AgentJSONIntakeProtocol, config.DefaultIntakeOrigin)
+}
+
+// rebuildTransientComponents recreates only the components that need to change during a restart.
+//
+// Components recreated (transient):
+//   - destinationsCtx: New context for new transport connections
+//   - pipelineProvider: New pipeline with updated endpoints and configuration
+//   - launchers: New launchers connected to the new pipeline
+func (a *logAgent) rebuildTransientComponents(processingRules []*config.ProcessingRule, wmeta option.Option[workloadmeta.Component], integrationsLogs integrations.Component, fingerprintConfig types.FingerprintConfig) {
+	// create NEW destinations context
+	destinationsCtx := client.NewDestinationsContext()
+
+	pipelineProvider := pipeline.NewProvider(
+		a.config.GetInt("logs_config.pipelines"),
+		a.auditor,
+		a.diagnosticMessageReceiver,
+		processingRules,
+		a.endpoints,
+		destinationsCtx,
+		NewStatusProvider(),
+		a.hostname,
+		a.config,
+		a.compression,
+		a.config.GetBool("logs_config.disable_distributed_senders"), // legacy
+		false, // serverless
+	)
+
+	// recreate launchers with new pipelineProvider
+	// use OLD: sources, auditor, tracker
+	lnchers := launchers.NewLaunchers(a.sources, pipelineProvider, a.auditor, a.tracker)
+	a.addLauncherInstances(lnchers, wmeta, integrationsLogs, fingerprintConfig)
+
+	// update agent with new components
+	// schedulers, sources, tracker, auditor, diagnosticMessageReceiver remain unchanged
+	a.destinationsCtx = destinationsCtx
+	a.pipelineProvider = pipelineProvider
+	a.launchers = lnchers
+}
+
+func (a *logAgent) addLauncherInstances(launchers *launchers.Launchers, wmeta option.Option[workloadmeta.Component], integrationsLogs integrations.Component, fingerprintConfig types.FingerprintConfig) {
```

Comment:
FYI the reason this was probably called `lnchrs` is to avoid the package name conflict with `github.com/DataDog/datadog-agent/pkg/logs/launchers`. 

I don't think this is hurting anything, but ocassionally the naming conflict can cause annoying compilation errors. Up to you if you want to keep this, just wanted to call it out